### ISSUE TITLE
rect=True with imgz=640 but in the training logs, it shows Size set to 384

### ISSUE THREAD

**Issue by ma7555** [ultralytics/ultralytics#15790]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

i have read how rect=True is supposed to work, however, it should still maintain the same imgz Size.

### Additional

with rect=False (commented)
```
model.train(
    data=r"dataset\dataset.yaml",
    cache=False,
    device=0,
    epochs=50,
    imgsz=640,
    batch=50,
    workers=8,
    pretrained=True,
    save_period=10,
    project="yolov8n-seg",
    name="yolov8n-seg",
    optimizer="AdamW",
    augment=False,
    exist_ok=True,
    degrees=30,
    fliplr=0,
    flipud=0.3,
    mosaic=0.5,
    crop_fraction=0.5,
    # rect=True,
    warmup_epochs=1
)
```
      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       1/50      8.09G     0.3605     0.4944      1.049     0.9128         22        640: 100%|██████████| 835/835 [05:18<00:00,  2.62it/s]

however, when setting rect=True, the Size in the logs changes to 384. Is this normal?

**Comment by Y-T-G** [comment]
`rect=True` will create a batch with minimum rectangle if all the images in the batch are of the same shape. So it will not pad it to `640x640`, but will resize to `384x640` if your original images are in 16:9 aspect ratio.

The `Size` in the log is the resized image width in the batch:
https://github.com/ultralytics/ultralytics/blob/d27664216b3daa514e959eed73db122ff3750b25/ultralytics/engine/trainer.py#L415

**Comment by ma7555** [comment]
@Y-T-G 
ok, so with rect=True the training actually happens on non-square input? because as of my readings it seemed that images still get padded to square but in 1 shot (like all over the batch at once instead of padding each image alone).

**Comment by Y-T-G** [comment]
It will be padded to square only if the images in the batch have different shapes. Otherwise, it's padded until the smallest side becomes divisible by 32.
